src/main/java/seedu/diary/ui/InternshipApplicationDetail.java

```diff
@@ -91,5 +94,8 @@ public InternshipApplicationDetail(InternshipApplication internshipApplication)
         interviewListPanel = new InterviewListPanel(
             FXCollections.observableArrayList(internshipApplication.getInterviews()));
         interviewListPanelPlaceHolder.getChildren().add(interviewListPanel.getRoot());
+        // initiate listener to InternshipApplication
+        internshipApplication.addPropertyChangeListener(DISPLAYED_INTERNSHIP_DETAIL, interviewListPanel);
```

Comment:
As interviewListPanel is not effectively a singleton, and InternshipApplication is persistent, there is a need to remove it from property change listener in order to avoid memory leak.

Reply:
As in remove interviewListPanel when it's destroyed.

Reply:
Okay, i will create a method to delete all listeners from IA before adding new PCL